app/controllers/services_controller.rb

```diff
@@ -1,147 +1,20 @@
-    end
-  end
-
-  def destroy
-    delete_response = cmr_client.delete_service(@provider_id, @native_id, token)
-
-    if delete_response.success?
-      flash[:success] = I18n.t('controllers.services.destroy.flash.success')
-      Rails.logger.info("Audit Log: Service #{@concept_id} with native_id #{@native_id} was deleted for #{@provider_id} by #{session[:urs_uid]}")
-
-      redirect_to service_revisions_path(id: delete_response.body['concept-id'], revision_id: delete_response.body['revision-id'])
-    else
-      Rails.logger.error("Delete Service Error: #{delete_response.inspect}")
-      Rails.logger.info("User #{current_user.urs_uid} attempted to delete Service #{@concept_id} with native_id #{@native_id} in provider #{@provider_id} but encountered an error.")
-
-      set_preview
-
-      flash[:error] = delete_response.error_message(i18n: I18n.t('controllers.services.destroy.flash.error'))
-      render :show
-    end
-  end
-
-  def revisions
-    add_breadcrumb breadcrumb_name(@service, 'services'), service_path(@concept_id)
-    add_breadcrumb 'Revision History', service_revisions_path(@concept_id)
-  end
-
-  def revert
-    latest_revision_id = @revisions.first['meta']['revision-id']
-
-    # Ingest revision
-    ingested_response = cmr_client.ingest_service(@service.to_json, @provider_id, @native_id, token)
-
-    if ingested_response.success?
-      flash[:success] = I18n.t('controllers.services.revert.flash.success')
-      Rails.logger.info("Audit Log: Service Revision for record #{@concept_id} with native_id: #{@native_id} for provider: #{@provider_id} by user #{session[:urs_uid]} has been successfully revised")
-      redirect_to service_revisions_path(revision_id: latest_revision_id.to_i + 1)
-    else
-      Rails.logger.error("Ingest (Revert) Service Error: #{ingested_response.inspect}")
-      Rails.logger.info("User #{current_user.urs_uid} attempted to revert Service #{@concept_id} by ingesting a previous revision in provider #{current_user.provider_id} but encountered an error.")
-
-      @errors = generate_ingest_errors(ingested_response)
-      flash[:error] = ingested_response.error_message(i18n: I18n.t('controllers.services.revert.flash.error'))
-      render action: 'revisions'
-    end
-  end
-
-  def download_json
-    send_data @service.to_json, type: 'application/json; charset=utf-8', disposition: "attachment; filename=#{@concept_id}.json"
+    super
   end
 
   private
 
-  def set_schema
-    @schema = UmmJsonSchema.new('services', 'umm-s-json-schema.json')
-    @schema.fetch_references(@schema.parsed_json)
-  end
-
-  def set_preview
-    @preview = UmmPreview.new(
-      schema_type: 'service',
-      preview_filename: 'umm-s-preview.json',
-      data: @service
-    )
-  end
-
-  def ensure_correct_service_provider
-    return if current_provider?(@provider_id)
-
-    set_record_action
-
-    set_user_permissions
-
-    set_preview
-
-    render :show
+  def resource_schema
```

Comment:
I was thinking we should declare this as a private constant (see http://www.chrisrolle.com/en/blog/private-constants-in-ruby), and have the whole file name. This works but I think is a little less clear.

Reply:
If we create a private constant then it's not possible to reference the constant anymore within the base class. 

Also there are 3 files names that are slightly different from each other. If you want the whole file name then we are going to need 3 different names.